### PR TITLE
Add support for SMTP email backend to Docker configuration

### PR DESCRIPTION
**Pull request by justindbelanger**
#### Any background context you want to provide?
The Docker image currently only supports the AWS SES
email backend. We (OPEN) would like the flexibility to use
SMTP for sending email.

#### What's this PR do?
This PR changes the Docker config file so that
it allows configuring different values for the email backend
that Django will use. It also copies the values from
Django's SMTP-related environment variables into memory.
(See the [Django docs for the full list](https://docs.djangoproject.com/en/3.0/topics/email/#smtp-backend) of available configuration variables.)

#### How should this be manually tested?
1. Build a new version of the Docker image.
2. Customize the docker-compose config file
    to include values for the following environment variables.
  - `DJANGO_EMAIL_BACKEND=django.core.mail.backends.smtp.EmailBackend`
  - `SERVER_EMAIL=seed_server@example.com`
  - `EMAIL_HOST=smtp.example.com`
  - `EMAIL_PORT=587`
  - `EMAIL_USE_TLS=True` _or_ `EMAIL_USE_SSL=True`
  - `EMAIL_HOST_USER=seed_account_support@example.com`
  - `EMAIL_HOST_PASSWORD=password123`
3. Run the Docker image using docker-compose.
4. Log in to the Web application.
5. Create or invite a new user with a real email address to join an organization
    in SEED.
6.  Verify that the user receives an email about their new account or inviting them to a new organization.

### CODE REVIEW

**Comment by nllong** [comment]
Hi @justindbelanger -- thanks for this PR. Can you update this code with latest develop? We just fixed the running (rather the not running) of coveralls when the env var is not set. That should make these tests pass. Thanks!

**Comment by justindbelanger** [comment]
> Hi @justindbelanger -- thanks for this PR. Can you update this code with latest develop? We just fixed the running (rather the not running) of coveralls when the env var is not set. That should make these tests pass. Thanks!

Thanks for the heads-up. I thought I'd broken something :sweat_smile:

**Comment by nllong** [comment]
Yeah, your changes seemed just fine. Hopefully the tests will pass now 🤞

File: config/settings/docker.py [hunk 1]
```diff
@@ -14,7 +14,9 @@
 # The optional vars will set the SERVER_EMAIL information as needed
 OPTIONAL_ENV_VARS = ['AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY', 'AWS_SES_REGION_NAME',
```
Comment (justindbelanger) [inline config/settings/docker.py:15]: Would it be worthwhile to separate the SMTP-related env vars out into another list, and then concatenate them with the rest of the optional env vars?
I'm thinking something like this:
``` python3
SMTP_ENV_VARS = ['EMAIL_HOST', 'EMAIL_PORT', 'EMAIL_HOST_USER',
                 'EMAIL_HOST_PASSWORD', 'EMAIL_USE_TLS', 'EMAIL_USE_SSL']

OPTIONAL_ENV_VARS = ['AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY', 'AWS_SES_REGION_NAME',
                     'AWS_SES_REGION_ENDPOINT', 'SERVER_EMAIL', 'SENTRY_JS_DSN', 'SENTRY_RAVEN_DSN',
                     'REDIS_PASSWORD', 'DJANGO_EMAIL_BACKEND'] + SMTP_ENV_VARS
```

File: config/settings/docker.py [hunk 2]
```diff
@@ -37,7 +39,7 @@
 # By default we are using SES as our email client. If you would like to use
 # another backend (e.g. SMTP), then please update this model to support both and
 # create a pull request.
-EMAIL_BACKEND = 'django_ses.SESBackend'
+EMAIL_BACKEND = (DJANGO_EMAIL_BACKEND if 'DJANGO_EMAIL_BACKEND' in os.environ else "django_ses.SESBackend")
```
Comment (nllong) [inline config/settings/docker.py:42]: thanks for removing the hard coded dependency!

**Review by nllong — APPROVED** [review]

File: config/settings/docker.py (hunk 1 above)
Comment (nllong) [reply config/settings/docker.py:15]: I think this would be a nice change.

Comment (justindbelanger) [reply config/settings/docker.py:20]: Done!

**Comment by justindbelanger** [comment]
Hey @nllong , I don't have permission to merge this branch in. If these changes look all right, could you please hit the "Merge" button for me?
Thanks for taking the time to look over this PR!

**Comment by nllong** [comment]
Looks good, thanks!

**Pull request merged**